db/wal_edit.cc

```diff
@@ -143,7 +143,7 @@ Status WalSet::AddWals(const WalAdditions& wals) {
 }
 
 Status WalSet::DeleteWal(const WalDeletion& wal) {
-  auto it = wals_.lower_bound(wal.GetLogNumber());
+  auto it = wals_.find(wal.GetLogNumber());
   // The WAL must exist and has been closed.
   if (it == wals_.end() || it->first != wal.GetLogNumber()) {
```

Comment:
You can also remove `it->first != wal.GetLogNumber()` from here (since it's only necessary with `lower_bound`).